lab_5_scrapper/scrapper.py

```diff
@@ -120,45 +247,106 @@ def __init__(self, full_url: str, article_id: int, config: Config) -> None:
         """
         Initializes an instance of the HTMLParser class
         """
-        pass
+        self._full_url = full_url
+        self._article_id = article_id
+        self._config = config
+        self.article = Article(self._full_url, self._article_id)
 
     def _fill_article_with_text(self, article_soup: BeautifulSoup) -> None:
         """
         Finds text of article
         """
-        pass
+        for text in article_soup.find_all('p')[:-2]:
+            self.article.text += text.text + '\n'
 
     def _fill_article_with_meta_information(self, article_soup: BeautifulSoup) -> None:
         """
         Finds meta information of article
         """
-        pass
+        title = article_soup.find('h1', attrs={'itemprop': 'headline'})
+        if not title:
+            self.article.title = 'WITHOUT TITLE'
+        self.article.title = title.text
+
+        authors = article_soup.find_all('a', class_='article-info_articleInfoAuthor__W0ZnW')
+        if not authors:
+            self.article.author = ["NOT FOUND"]
+        self.article.author = [author.text for author in authors]
+
+        topics = article_soup.find_all("a", class_='article-tags_articleTagsLink__El86x')
+        if topics:
+            self.article.topics = [topic.text for topic in topics]
+
+        date = article_soup.find('span', attrs={'class': 'article-info_articleInfoDate__S0E0P'})
+        string_date = date.text
+        self.article.date = self.unify_date_format(string_date)
 
     def unify_date_format(self, date_str: str) -> datetime.datetime:
         """
         Unifies date format
         """
-        pass
+        list_date = date_str.lower().split()
+
+        months_collection = {"января": "01", "февраля": "02", "марта": "03",
+                             "апреля": "04", "мая": "05", "июня": "06",
+                             "июля": "07", "августа": "08", "сентября": "09",
+                             "октября": "10", "ноября": "11", "декабря": "12"}
+
+        year = '2023'
+        month = ''
+        day = ''
+        time = ''
+
+        for element in list_date:
+            if ':' in element:
+                time += element
+            if element in months_collection:
+                month += months_collection[element]
+            digits = ['1', '2', '3', '4', '5', '6', '7', '8', '9']
+            if element.isdigit() and element in digits:
+                day += '0' + element
+            if element.isdigit() and element not in digits:
+                day += element
+
+        correct_date = year + '-' + month + '-' + day + ' ' + time
+        return datetime.datetime.strptime(correct_date, '%Y-%m-%d %H:%M')
```

Comment:
can you explain why it is necessary to perform all this preprocessing before using `strptime`? is there an easier way to convert one date format into another without an intermediate format?